src/tv2-common/actions/executeAction.ts

```diff
@@ -1928,7 +1938,16 @@ async function executeActionFadeDownPersistedAudioLevels<
 			timelineObjects: []
 		}
 	}
-	context.insertPiece('current', resetSisyfosPersistedLevelsPiece)
+	await context.insertPiece('current', resetSisyfosPersistedLevelsPiece)
+}
+
+async function executeActionSelectRobotPreset(context: ITV2ActionExecutionContext, preset: number): Promise<void> {
+	const telemetricsPiece: IBlueprintPiece<PieceMetaData> = createTelemetricsPiece(
+		`selectRobotPreset${preset}`,
+		preset,
+		'now'
+	) as IBlueprintPiece<PieceMetaData>
+	await context.insertPiece('current', telemetricsPiece)
```

Comment:
In theory, yes. But wouldn't we face this risk no matter how we calculate the start time of the piece?
Do you think we should guard against it?

Reply:
I don't think there is much we can do except other than having a separate source layer and a separate mapping for robot/telemetrics adlib actions, but this seems like something that would be extremely rare, so maybe we shouldn't spend time on it unless solving it becomes an explicit requirement.
@julusian could you confirm that a pieceInstance inserted 'now' can theoretically collide with an existing pieceInstance (covering it completely if the start times happen to be identical by accident), and there is no magic around it?

Reply:
yeah there is a chance that it could completely block the other, or much more likely that it will be inserted just before or after the other, causing one of them to run for a few hundred before the other takes over.